brain/python/arrayHelpers.cpp

```diff
@@ -239,6 +239,52 @@ bp::object toNumpy( const brain::Matrix4f& matrix )
     return bp::object( bp::handle<>( array ));
 }
 
+
+bp::object frameToTuple(CompartmentReportFrame&& frame)
+{
+    return bp::make_tuple(frame.getTimestamp(), toNumpy(frame.takeData()));
+}
+
+bp::object framesToTuple(CompartmentReportFrames&& frames)
+{
+    if(frames.empty())
+        return bp::make_tuple(-1.f,std::vector<float>());
+    if(frames.size() == 1)
+        return frameToTuple(std::move(frames.front()));
+
+    std::vector<double> timestamps;
+    timestamps.reserve(frames.size());
+    for (const auto& frame : frames)
+        timestamps.push_back(frame.getTimestamp());
+
+    size_t size = frames.size() * frames[0].getData().size();
+
+    npy_intp dimensions[2] = {(npy_intp)frames.size(),
+                              (npy_intp)frames[0].getData().size()};
+    float* matrixData = (float*)malloc(sizeof(float) * size);
+    if (!matrixData)
+    {
+        PyErr_SetString(PyExc_MemoryError,
+                        "Allocating numpy array for Matrix4f");
+        bp::throw_error_already_set();
+    }
+
+    for (const auto& frame : frames)
+    {
+        std::copy(frame.getData().begin(), frame.getData().end(), matrixData);
+        matrixData += frame.getData().size();
+    }
+
+    PyObject* array =
+        PyArray_New(&PyArray_Type, 2, dimensions, NPY_FLOAT, 0, matrixData, 0,
+                    NPY_ARRAY_OWNDATA | NPY_ARRAY_F_CONTIGUOUS, 0);
```

Comment:
I think the layout must be NPY_ARRAY_C_CONTIGUOUS. Furthermore, in the tests you check the shape of the ouput numpy array, but there's no test for the values themselves (except for the non matrix case).
I even think you can use PyArray_SimpleNewFromData, which has fewer parameters.